TheThingsNetwork/lorawan-stack
# Organization `fanout_notifications` option

#### Summary



Closes https://github.com/TheThingsNetwork/lorawan-stack/issues/6751

This the OS counterpart of: https://github.com/TheThingsIndustries/lorawan-stack/pull/4006

#### Changes



- Add migration
  - Add boolean column `fanout_notifications` to organizations table.
  - Make all rows of table have its value as `true`, preserves old behavior.
- Add `FanoutNotifications` field to organization API definition.
- Update notification registry to take into consideration the `FantoutNotifications` field from the `organization` entity.
- Update unit tests for new field in `organization` store.
- Add unit tests for `notification_registry`, revolves around the scenario where an `organization` is an `(admin/tech)_contact`. 

#### Testing



Unit tests
<details>
<summary> Test steps </summary>

1. Create two users. (could be the admin user and an extra user)
2. Create an organization
2.1. Add both as members.
2.2. Add one of the users as the administrative contact
3. Create an application and set the organization as its administrative contact.
4. Create an API-KEY for the application.
4.1. There should be only an email for the administrative contact of the organization.
5. Update organization and set `fanout_notifications` to true. Use `ttn-lw-cli org set org-1 --fanout-notifications=true`
6. Create another API-KEY for the application.
6.1. There should be one email for each member of the organization.

Observation:
When testing it locally it is usefull to have `dir` as the email provider. Can be done by setting the Stack config to:
```yaml
is:
  email:
    dir: ".env/emails"
    provider: "dir"
    sender-address: "nicholas.cristofaro@thethingsindustries.com"
    sender-name: "nicholas-cristofaro"

```

</details>


##### Regressions




#### Notes for Reviewers



1. This does not include changes to the front-end so it only references the issue, will update the issue after this is approved.
2. Will also update the docs after the PR is approved and no longer subject to changes.

#### Checklist
